rets/http/client.py

```diff
@@ -286,31 +292,54 @@ def _url_for(self, transaction: str) -> str:
             raise RetsClientError('No URL found for transaction %s' % transaction)
         return urljoin(self._base_url, url)
 
-    def _http_post(self, url: str, headers: dict = None, payload: dict = None) -> Response:
+    def _http_request(self, url: str, headers: dict = None, payload: dict = None) -> Response:
+        if self._use_get_method:
+            return self._http_get(url, headers, payload)
+        return self._http_post(url, headers, payload)
+
+    def _http_get(self, url: str, headers: dict = None, payload: dict = None) -> Response:
         if not self._session:
             raise RetsClientError('Session not instantiated. Call .login() first')
 
-        if headers is None:
-            headers = {}
-        else:
-            headers = headers.copy()
-        headers.update({
+        request_headers = {
+            **(headers or {}),
             'User-Agent': self.user_agent,
             'RETS-Version': self.rets_version,
-        })
+            'RETS-UA-Authorization': self._rets_ua_authorization()
+        }
 
-        if self._http_auth:
-            response = self._session.post(url, auth=self._http_auth, headers=headers, data=payload)
-        else:
-            headers['RETS-UA-Authorization'] = self._user_agent_auth_digest()
-            response = self._session.post(url, headers=headers, data=payload)
+        # If payload exists, convert to query string and append to url.
+        if payload:
+            url = '%s?%s' % (url, urlencode(payload))
 
+        response = self._session.get(url, headers=request_headers, auth=self._http_auth)
```

Comment:
looks like apart from the `if payload` part and getting this `response`, everything is the same in `_http_get` and `_http_post` - you could probably just have one method and differentiate this part based on if we want to use GET or POST